### ISSUE TITLE
Provide access to real `spawn2` process when launched with P_2_THREADSAFE

### ISSUE THREAD

**Issue by dmik** [bitwiseworks/libcx#51]
This is a continuation of #49. When P_2_THREADSAFE is used, a wrapper process is executed which then starts the real child process. Now the calling process will get a PID of the wrapper process rather than of the real child process (which will be its grandchild, in fact). This is generally fine but has some limitations (e.g. the starting process won't be able to give memory to the grand-child with `DosGIveSharedMem` since it doesn't know its PID).

The grand-child PID is in fact already stored in internal LIBCx structures. The reason it's not returned by `spawn2` now is because it's impossible to wait on a grand-child process directly, so `wait` and `waitpid` APIs wouldn't work. What needs to be done is override those `wait` APIs and allow them to accept both the wrapper PID and the real child PID replacing the latter with the wrapper PID to make the wait function work. Not very needed now so postponed.

**Comment by dmik** [comment]
This was actually closed by 6832aaf9d328a643aed9853902629c33c44b46d5 and subsequent fixes.
